corehq/apps/userreports/indicators/specs.py

```diff
@@ -97,3 +97,18 @@ class ChoiceListIndicatorSpec(PropertyReferenceIndicatorSpecBase):
 
     def get_operator(self):
         return IN_MULTISELECT if self.select_style == 'multiple' else EQUAL
+
+
+class LedgerBalancesIndicatorSpec(IndicatorSpecBase):
+    type = TypeProperty('ledger_balances')
+    product_codes = ListProperty(required=True)
+    ledger_section = StringProperty()
+    case_id_expression = DictProperty()
+
+    def get_case_id_expression(self):
+        from corehq.apps.userreports.expressions.factory import ExpressionFactory
+        expression_spec = self.case_id_expression or {
+            "type": "property_name",
+            "property_name": "_id",
+        }
+        return ExpressionFactory.from_spec(expression_spec)
```

Comment:
I think I disagree. My general philosophy for UCR has to push all the config into the json and not have any surprising magic happen in the framework. If we want to provide a default I'd advocate for that default being specified either in the UI or on save, so that everything that deals with data is fully specified in the self-contained definition.